src/sql/workbench/api/browser/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -721,7 +722,8 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 				if (result) {
 					if (result.previous.scheme === Schemas.untitled) {
 						let untitledNbName: URI = URI.parse(`untitled:${path.basename(result.previous.path)}`);
-						this.doOpenEditor(untitledNbName, { initialContent: fs.readFileSync(result.previous.path).toString(), initialDirtyState: false });
+						let content = await this.fileService.readFile(URI.parse(result.previous.path));
+						this.doOpenEditor(untitledNbName, { initialContent: content.value.toString(), initialDirtyState: false });
```

Comment:
doOpenEditor returns a promise

Reply:
nothing new I added here. Anyway we're not using the returned value and this is the last operation.

Reply:
It's important in this scenario, because you don't guarantee that the notebook will be open by the time this method ends. doOpenEditor will be scheduled to run on the either the next or end of current poll - rather than during the execution of this method.

It creates issues like the one in #7064 

It's either something we need to solve in this PR or create another issue for it.

Reply:
Is that really what's causing those problems? Why does it matter if the notebook is actually opened when this function finishes?

A small potential issue is that if the switching takes a while then quickly clicking "next" or "previous" could cause issues - but that's separate I'd think.

Reply:
I think as a general rule of thumb, if we're using functions that return a promise we should either await, .then, or return them.

I'm not saying this specific one is causing issues, but not following that rule can cause regressions down the line. 

I'm currently looking for an eslint rule to warn when you don't await these functions.